src/main/scala/rps/Game.scala

```diff
@@ -21,38 +21,45 @@ object Game {
 
   def play(): Unit = {
     println(s"""Choose your weapon:
-    ${Weapon.Rock} - 🪨 Rock
-    ${Weapon.Paper} - 📄 Paper
-    ${Weapon.Scissors} - ✂️ Scissors
+    ${Weapon.Rock} - ${matchWeapon(Weapon.Rock)}
+    ${Weapon.Paper} - ${matchWeapon(Weapon.Paper)}
+    ${Weapon.Scissors} - ${matchWeapon(Weapon.Scissors)}
     """)
     val rawUserInput = readLine()
     val userWeapon = rawUserInput
     val cpuWeapon = generateComputerMove()
 
     val result = checkWinner(userWeapon, cpuWeapon)
 
-    println(s"You chose: ${userWeapon} | your opponent chose: ${cpuWeapon}")
+    println(s"You chose: ${matchWeapon(userWeapon)} | your opponent chose: ${matchWeapon(cpuWeapon)}")
     println(announceWinner(result))
   }
 
+  def matchWeapon(weapon: String): String = {
+    weapon match {
+      case Weapon.Rock     => "🪨 Rock"
+      case Weapon.Paper    => "📄 Paper"
+      case Weapon.Scissors => "✂️ Scissors"
+      case _               => "💩 An invalid weapon"
+    }
+  }
+
   def announceWinner(result: String): String = {
-    if (result == Winner.Draw) "It's a draw ✍️"
-    else if (result == Winner.User) s"You win 🎉"
-    else "You lose 🤷"
+    result match {
+      case Winner.Draw => "It's a draw ✍️"
+      case Winner.User => "You win 🎉"
+      case _           => "You lose 🤷"
+    }
   }
 
-  def checkWinner(user: String, cpu: String): String = {
-    if (user == cpu) {
-      Winner.Draw
-    } else if (
-      (user == Weapon.Rock && cpu == Weapon.Scissors)
-      || (user == Weapon.Paper && cpu == Weapon.Rock)
-      || (user == Weapon.Scissors && cpu == Weapon.Paper)
-    ) {
-      Winner.User
-    } else {
-      // also covers the case for an invalid weapon (e.g. user inputs "4")
-      Winner.Cpu
+  def checkWinner(userMove: String, computerMove: String): String = {
+    (userMove, computerMove) match {
+      case (x, y) if (x == y)              => Winner.Draw
```

Comment:
```suggestion
      case (x, y) if x == y              => Winner.Draw
```
le parentesi non servono qui